src/Ocelot/LoadBalancer/LoadBalancers/CookieStickySessions.cs

```diff
@@ -1,85 +1,81 @@
 using Microsoft.AspNetCore.Http;
 using Ocelot.Infrastructure;
+using Ocelot.Middleware;
 using Ocelot.Responses;
 using Ocelot.Values;
 
-namespace Ocelot.LoadBalancer.LoadBalancers
+namespace Ocelot.LoadBalancer.LoadBalancers;
+
+public class CookieStickySessions : ILoadBalancer
 {
-    public class CookieStickySessions : ILoadBalancer
+    private readonly int _keyExpiryInMs;
+    private readonly string _cookieName;
+    private readonly ILoadBalancer _loadBalancer;
+    private readonly IBus<StickySession> _bus;
+
+    private static readonly object _lock = new();
+    private static readonly ConcurrentDictionary<string, StickySession> _stored = new(); // TODO Inject instead of static sharing
+
+    public CookieStickySessions(ILoadBalancer loadBalancer, string cookieName, int keyExpiryInMs, IBus<StickySession> bus)
     {
-        private readonly int _keyExpiryInMs;
-        private readonly string _key;
-        private readonly ILoadBalancer _loadBalancer;
-        private readonly ConcurrentDictionary<string, StickySession> _stored;
-        private readonly IBus<StickySession> _bus;
-        private readonly object _lock = new();
+        _bus = bus;
+        _cookieName = cookieName;
+        _keyExpiryInMs = keyExpiryInMs;
+        _loadBalancer = loadBalancer;
+        _bus.Subscribe(CheckExpiry);
+    }
 
-        public CookieStickySessions(ILoadBalancer loadBalancer, string key, int keyExpiryInMs, IBus<StickySession> bus)
+    private void CheckExpiry(StickySession sticky)
```

Comment:
Instead of mixing ```Lock``` and ```ConcurrentDictionary``` why not use the lock only, with a plain and simple dictionary eg.
```c#
private void CheckExpiry(StickySession sticky)
    {
        // TODO Get test coverage for this
        lock (Lock)
        {
            if (!Stored.TryGetValue(sticky.Key, out var session))
            {
                return;
            }

            if (session.Expiry >= DateTime.UtcNow)
            {
                return;
            }

            Stored.Remove(session.Key);
            _loadBalancer.Release(session.HostAndPort);
        }
    }
```